diff -aurN totem-1.5.92.orig/configure.in totem-1.5.92/configure.in
--- totem-1.5.92.orig/configure.in	2006-08-22 11:00:07.000000000 +0200
+++ totem-1.5.92/configure.in	2006-08-22 20:04:33.000000000 +0200
@@ -127,36 +127,6 @@
 	dnl Check for required plugins
 	gst010_toolsdir=`$PKG_CONFIG --variable=toolsdir gstreamer-0.10`
 	gst010_inspect="$gst010_toolsdir/gst-inspect-0.10"
-
-	dnl Check for elements from gst-plugins-base
-	for base_element in playbin ffmpegcolorspace videoscale
-	do
-		AC_MSG_CHECKING([GStreamer 0.10 $base_element plugin])
-		if $gst010_inspect $base_element >/dev/null 2>/dev/null; then
-			AC_MSG_RESULT([yes])
-		else
-			AC_MSG_RESULT([no])
-			AC_MSG_ERROR([
-				Cannot find required GStreamer-0.10 plugin '$base_element'.
-				It should be part of gst-plugins-base. Please install it.
-			])
-		fi
-	done
-
-	dnl Check for elements from gst-plugins-good
-	for good_element in gconfaudiosink gconfvideosink goom
-	do
-		AC_MSG_CHECKING([GStreamer 0.10 $good_element plugin])
-		if $gst010_inspect $good_element >/dev/null 2>/dev/null; then
-			AC_MSG_RESULT([yes])
-		else
-			AC_MSG_RESULT([no])
-			AC_MSG_ERROR([
-				Cannot find required GStreamer-0.10 plugin '$good_element'.
-				It should be part of gst-plugins-good. Please install it.
-			])
-		fi
-	done
 fi
 
 AM_CONDITIONAL(TOTEM_GST, test "x$HAVE_GSTREAMER" = "xyes")
