diff -aurN totem-1.5.1.orig/configure.in totem-1.5.1/configure.in
--- totem-1.5.1.orig/configure.in	2006-05-17 15:13:28.000000000 +0200
+++ totem-1.5.1/configure.in	2006-06-08 23:51:56.630289440 +0200
@@ -130,40 +130,6 @@
 
 	AC_MSG_RESULT(GStreamer-$GST_MAJORMINOR)
 
-	dnl Check for required plugins
-	gst010_toolsdir=`$PKG_CONFIG --variable=toolsdir gstreamer-0.10`
-	gst010_inspect="$gst010_toolsdir/gst-inspect-0.10"
-
-	dnl Check for elements from gst-plugins-base
-	for base_element in playbin ffmpegcolorspace videoscale
-	do
-		AC_MSG_CHECKING([GStreamer 0.10 $base_element plugin])
-		if $gst010_inspect $base_element >/dev/null 2>/dev/null; then
-			AC_MSG_RESULT([yes])
-		else
-			AC_MSG_RESULT([no])
-			AC_MSG_ERROR([
-				Cannot find required GStreamer-0.10 plugin '$base_element'.
-				It should be part of gst-plugins-base. Please install it.
-			])
-		fi
-	done
-
-	dnl Check for elements from gst-plugins-good
-	for good_element in gconfaudiosink gconfvideosink
-	do
-		AC_MSG_CHECKING([GStreamer 0.10 $good_element plugin])
-		if $gst010_inspect $good_element >/dev/null 2>/dev/null; then
-			AC_MSG_RESULT([yes])
-		else
-			AC_MSG_RESULT([no])
-			AC_MSG_ERROR([
-				Cannot find required GStreamer-0.10 plugin '$good_element'.
-				It should be part of gst-plugins-good. Please install it.
-			])
-		fi
-	done
-
 	dnl We want at least GLib-2.8 for the GStreamer-0.10 backend
 	AC_MSG_CHECKING([whether GLib has atomic refcounting for objects])
 	PKG_CHECK_MODULES(GLIB_WITH_ATOMIC_REFCOUNTING, glib-2.0 >= 2.8.0, [
