diff -urN totem-2.30.1/configure.in totem-2.30.1.new/configure.in
--- totem-2.30.1/configure.in	2010-04-27 11:17:40.000000000 +0200
+++ totem-2.30.1.new/configure.in	2010-04-27 15:02:46.203562788 +0200
@@ -101,52 +101,6 @@
 AC_SUBST(GST_LIBS)
 EXTRA_BACKEND_LIBS="-lgstinterfaces-$GST_MAJORMINOR -lgstvideo-$GST_MAJORMINOR -lgstaudio-$GST_MAJORMINOR -lgstpbutils-$GST_MAJORMINOR"
 
-dnl Check for required plugins
-gst010_toolsdir=`$PKG_CONFIG --variable=toolsdir gstreamer-0.10`
-gst010_inspect="$gst010_toolsdir/gst-inspect-0.10"
-
-dnl Give error and exit if we don't have the gst_inspect tool
-AC_MSG_CHECKING([GStreamer 0.10 inspection tool])
-if test -r "$gst010_inspect"; then
-	AC_MSG_RESULT([yes])
-else
-	AC_MSG_RESULT([no])
-	AC_MSG_ERROR([
-		Cannot find required GStreamer-0.10 tool 'gst-inspect-0.10'.
-		It should be part of gstreamer-0_10-utils. Please install it.
-	])
-fi
-
-dnl Check for elements from gst-plugins-base
-for base_element in playbin2 ffmpegcolorspace videoscale
-do
-	AC_MSG_CHECKING([GStreamer 0.10 $base_element plugin])
-	if $gst010_inspect $base_element >/dev/null 2>/dev/null; then
-		AC_MSG_RESULT([yes])
-	else
-		AC_MSG_RESULT([no])
-		AC_MSG_ERROR([
-			Cannot find required GStreamer-0.10 plugin '$base_element'.
-			It should be part of gst-plugins-base. Please install it.
-		])
-	fi
-done
-
-dnl Check for elements from gst-plugins-good
-for good_element in gconfaudiosink gconfvideosink goom
-do
-	AC_MSG_CHECKING([GStreamer 0.10 $good_element plugin])
-	if $gst010_inspect $good_element >/dev/null 2>/dev/null; then
-		AC_MSG_RESULT([yes])
-	else
-		AC_MSG_RESULT([no])
-		AC_MSG_ERROR([
-			Cannot find required GStreamer-0.10 plugin '$good_element'.
-			It should be part of gst-plugins-good. Please install it.
-		])
-	fi
-done
-
 dnl Check the smclient backend
 GDK_TARGET="$($PKG_CONFIG --variable target gdk-2.0)"
 
@@ -551,16 +505,6 @@
 			if test "${HAVE_LIBGDATA_0_7}" = "yes" ; then
 				AC_DEFINE([HAVE_LIBGDATA_0_7],[1],[Define if libgdata >= 0.7.0 is available])
 			fi
-
-			dnl We need the souphttpsrc element for the YouTube plugin
-			AC_MSG_CHECKING([GStreamer 0.10 souphttpsrc plugin])
-			if $gst010_inspect souphttpsrc >/dev/null 2>/dev/null; then
-				AC_MSG_RESULT([yes])
-			else
-				AC_MSG_RESULT([no])
-				plugin_error_or_ignore "you need souphttpsrc from gst-plugins-good installed for the YouTube plugin"
-				add_plugin="0"
-			fi
 		;;
 	esac
 
