diff -p -up totem-2.23.4/src/backend/Makefile.am~ totem-2.23.4/src/backend/Makefile.am
--- totem-2.23.4/src/backend/Makefile.am~	2008-03-04 20:39:41.000000000 +0100
+++ totem-2.23.4/src/backend/Makefile.am	2008-07-04 11:33:02.000000000 +0200
@@ -18,7 +18,7 @@ bvw_test_CFLAGS = \
 
 bvw_test_LDADD = \
 	libbaconvideowidget.la \
-	$(EXTRA_GNOME_LIBS) $(GTK_LIBS) $(XVIDMODE_LIBS) $(NVTV_LIBS) $(XTEST_LIBS) $(X_LIBS)
+	$(EXTRA_GNOME_LIBS) $(XVIDMODE_LIBS) $(XTEST_LIBS)
 
 BVWMARSHALFILES = baconvideowidget-marshal.c baconvideowidget-marshal.h
 GLIB_GENMARSHAL=`pkg-config --variable=glib_genmarshal glib-2.0`
@@ -68,6 +68,9 @@ libbaconvideowidget_la_LIBADD =	\
 	$(XVIDMODE_LIBS)	\
 	$(MM_LIBS)		\
 	$(GST_LIBS)		\
+	$(NVTV_LIBS)		\
+	$(GTK_LIBS)		\
+	$(X_LIBS)		\
 	$(MISSING_PLUGINS_LIBS)
 
 libbaconvideowidget_la_LDFLAGS=	\
