diff -urN aa/totem-0.99.9/configure.in totem-0.99.9/configure.in
--- aa/totem-0.99.9/configure.in	2004-02-13 12:07:52.000000000 +0100
+++ totem-0.99.9/configure.in	2004-03-10 18:12:11.228976912 +0100
@@ -39,8 +39,8 @@
 dnl And we can also ask for the right version of gstreamer
 	HAVE_GSTREAMER=no
 
-dnl start with 0.7
-	GST_MAJORMINOR=0.7
+dnl start with 0.8
+	GST_MAJORMINOR=0.8
 	PKG_CHECK_MODULES(GST, \
 			gstreamer-play-$GST_MAJORMINOR >= $GST_REQS \
 			gstreamer-gconf-$GST_MAJORMINOR >= $GST_REQS \
