diff -urN aa/totem-0.99.15.1/data/totem.desktop.in.in.in totem-0.99.15.1/data/totem.desktop.in.in.in
--- aa/totem-0.99.15.1/data/totem.desktop.in.in.in	2004-07-22 18:00:38.000000000 +0200
+++ totem-0.99.15.1/data/totem.desktop.in.in.in	2004-07-24 13:34:38.337804936 +0200
@@ -7,9 +7,10 @@
 Terminal=false
 Type=Application
 DocPath=
-Categories=GNOME;Application;AudioVideo;Player;X-Ximian-Main;X-Red-Hat-Base;
+Categories=GTK;GNOME;AudioVideo;Player;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=totem
 X-GNOME-Bugzilla-Component=general
 X-GNOME-Bugzilla-Version=@VERSION@
 StartupNotify=true
+# vi: encoding=utf-8
