diff -aurN totem-1.0.orig/data/totem.desktop.in.in.in totem-1.0/data/totem.desktop.in.in.in
--- totem-1.0.orig/data/totem.desktop.in.in.in	2005-01-04 19:38:12.000000000 +0100
+++ totem-1.0/data/totem.desktop.in.in.in	2005-03-20 10:39:29.658296216 +0100
@@ -7,7 +7,7 @@
 Terminal=false
 Type=Application
 DocPath=
-Categories=GNOME;Application;AudioVideo;Player;X-Ximian-Main;X-Red-Hat-Base;
+Categories=GTK;GNOME;AudioVideo;Player;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=totem
 X-GNOME-Bugzilla-Component=general
