diff -u -r1.123 totem.glade
--- data/totem.glade	12 Jul 2006 09:50:24 -0000	1.123
+++ data/totem.glade	11 Sep 2006 10:46:43 -0000
@@ -2842,8 +2842,6 @@
 	      <child>
 		<widget class="GtkVBox" id="tmw_bvw_vbox">
 		  <property name="border_width">1</property>
-		  <property name="width_request">420</property>
-		  <property name="height_request">315</property>
 		  <property name="visible">True</property>
 		  <property name="homogeneous">False</property>
 		  <property name="spacing">0</property>
