--- totem-3.26.0/src/backend/meson.build.configfix	2018-03-17 22:39:59.380045123 +0100
+++ totem-3.26.0/src/backend/meson.build	2018-03-17 22:40:14.210790112 +0100
@@ -6,32 +6,17 @@ gst_inspect = find_program(
   required: false
 )
 
-assert(gst_inspect.found(),
-       'Cannot find required GStreamer-1.0 tool "gst-inspect-1.0". It should be part of gstreamer-1_0-utils. Please install it.')
-
 gst_base_plugins = [
   'playbin',
   'videoscale'
 ]
 
-foreach plugin: gst_base_plugins
-  r = run_command(gst_inspect, plugin)
-  assert(r.returncode() == 0,
-         'Cannot find required GStreamer-1.0 plugin "' + plugin + '". It should be part of gst-plugins-base. Please install it.')
-endforeach
-
 gst_good_plugins = [
   'autoaudiosink',
   'goom',
   'scaletempo'
 ]
 
-foreach plugin: gst_good_plugins
-  r = run_command(gst_inspect, plugin)
-  assert(r.returncode() == 0,
-         'Cannot find required GStreamer-1.0 plugin "' + plugin + '". It should be part of gst-plugins-good. Please install it.')
-endforeach
-
 sources = files(
   'bacon-time-label.c',
   'bacon-video-controls-actor.c',
