diff -aurN totem-1.2.1.orig/data/totem.desktop.in.in.in totem-1.2.1/data/totem.desktop.in.in.in
--- totem-1.2.1.orig/data/totem.desktop.in.in.in	2005-11-24 20:00:19.000000000 +0100
+++ totem-1.2.1/data/totem.desktop.in.in.in	2005-12-11 22:49:44.000000000 +0100
@@ -6,7 +6,7 @@
 Icon=media-player-48.png
 Terminal=false
 Type=Application
-Categories=GNOME;GTK;AudioVideo;Video;Player;X-Ximian-Main;X-Red-Hat-Base;
+Categories=GTK;GNOME;AudioVideo;Player;Video;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=totem
 X-GNOME-Bugzilla-Component=general
