--- totem-3.8.0/configure.ac.new	2013-03-28 21:32:33.215758291 +0100
+++ totem-3.8.0/configure.ac	2013-03-28 21:36:08.991256002 +0100
@@ -134,63 +134,6 @@ BACKEND_MODULES="gstreamer-1.0 >= $GST_R
 PKG_CHECK_MODULES(BACKEND, $BACKEND_MODULES)
 PKG_CHECK_MODULES(BACKEND_TEST, $BACKEND_MODULES gtk+-3.0 >= $GTK_REQS)
 
-dnl Check for required plugins
-gst10_toolsdir=`$PKG_CONFIG --variable=toolsdir gstreamer-1.0`
-gst10_inspect="$gst10_toolsdir/gst-inspect-1.0"
-
-dnl Give error and exit if we don't have the gst_inspect tool
-AC_MSG_CHECKING([GStreamer 1.0 inspection tool])
-if test -r "$gst10_inspect"; then
-	AC_MSG_RESULT([yes])
-else
-	AC_MSG_RESULT([no])
-	AC_MSG_ERROR([
-		Cannot find required GStreamer-1.0 tool 'gst-inspect-1.0'.
-		It should be part of gstreamer-1_0-utils. Please install it.
-	])
-fi
-
-dnl Check for elements from gst-plugins-base
-dnl Set plugins which contain below elements
-set -- playback videoscale
-for base_element in playbin videoscale
-do
-	AC_MSG_CHECKING([GStreamer 1.0 $base_element element])
-	if $gst10_inspect $base_element >/dev/null 2>/dev/null; then
-		AC_MSG_RESULT([yes])
-	else
-		AC_MSG_RESULT([no])
-		AC_MSG_ERROR([
-			Cannot find required GStreamer-1.0 plugin '$1'.
-			It should be part of gst-plugins-base. Please install it.
-		])
-
-	fi
-	shift;
-done
-
-dnl Check for elements from gst-plugins-good
-dnl Set plugins which contain below elements
-set -- autodetect goom scaletempo
-for good_element in autoaudiosink goom scaletempo
-do
-	AC_MSG_CHECKING([GStreamer 1.0 $good_element element])
-	if $gst10_inspect $good_element >/dev/null 2>/dev/null; then
-		AC_MSG_RESULT([yes])
-	else
-		AC_MSG_RESULT([no])
-		AC_MSG_ERROR([
-			Cannot find required GStreamer-1.0 plugin '$1'.
-			It should be part of gst-plugins-good. Please install it.
-		])
-	fi
-	shift;
-done
-
-dnl Check for elements from gst-plugins-bad
-dnl Set plugins which contain below elements
-PKG_CHECK_MODULES(GST_PLUGINS_BAD, gstreamer-plugins-bad-1.0 >= 1.0.2)
-
 dnl Check the smclient backend
 AC_MSG_CHECKING([which smclient backend to use])
 AC_ARG_WITH([smclient],
