diff -urN totem-1.2.0.orig/configure.in totem-1.2.0/configure.in
--- totem-1.2.0.orig/configure.in	2005-09-10 21:27:49.077593632 +0200
+++ totem-1.2.0/configure.in	2005-09-10 21:28:11.370204640 +0200
@@ -277,10 +277,6 @@
 if test "x$ENABLE_MOZILLA" = "xyes" ; then
 	dnl This only works on gecko 1.8
 	MOZILLA_IDLDIR="`$PKG_CONFIG --variable=idldir $MOZILLA-xpcom`"
-	dnl Fallback for older versions
-	if test "x$MOZILLA_IDLDIR" = "x"; then
-		MOZILLA_IDLDIR="`echo $MOZILLA_LIBDIR | sed -e s!lib!share/idl!`"
-	fi
 	dnl Some distributions (Gentoo) have it in unusual places
 	if test "x$MOZILLA_IDLDIR" = "x"; then
 		MOZILLA_IDLDIR="$MOZILLA_INCLUDE_ROOT/idl"
