diff -urN aa/totem-0.99.9/configure.in totem-0.99.9/configure.in
--- aa/totem-0.99.9/configure.in	2004-02-13 12:07:52.000000000 +0100
+++ totem-0.99.9/configure.in	2004-02-27 16:17:01.418998528 +0100
@@ -273,7 +273,7 @@
 fi
 
 dnl Add the languages which your application supports here.
-ALL_LINGUAS="am az bg be bn ca cs cy da de el en_CA en_GB es et fi fr ga he hr hu it ja ko lt lv ml ms nl no pl pt pt_BR ro ru sr sr@Latn sv tr uk vi zh_CN zh_TW"
+ALL_LINGUAS="am az bg be bn ca cs cy da de el en_CA en_GB es et fi fr ga he hr hu it ja ko lt lv ml ms nb nl pl pt pt_BR ro ru sr sr@Latn sv tr uk vi zh_CN zh_TW"
 AM_GLIB_GNU_GETTEXT
 
 AC_OUTPUT([
